From ab198c2458496f8c1c770f50dbfb662e6e496378 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Fri, 29 Oct 2021 12:37:25 +0300
Subject: [PATCH] [SW-0] gaudi2: arc: disable alignment checks on data memory

Currently, code is compiled with un-aligned data access for ARC as
ARC supports unaligned data access. This feature is not enabled for
ARC0 running pcie initialization code. This patch add changes to
disable alignment checks on data memory by setting AD bit to 1 in
STATUS32 register for ARC0.

Change-Id: Ic73367f4388beb8c612ba2fabffbfd55c7028d4b
---

diff --git a/arch/gaudi2/arc/secondary.S b/arch/gaudi2/arc/secondary.S
index 753af2c..0ea094d 100644
--- a/arch/gaudi2/arc/secondary.S
+++ b/arch/gaudi2/arc/secondary.S
@@ -9,4 +9,9 @@
 secondary_entry:
 	mov	sp, 0x27020400	/* stack top set to secondary CPU DCCM 1k offset */
 	mov	fp, sp
+
+	/* Disable un-aligned checks on data memory */
+	lr r0, [STATUS32]
+	bset r0, r0, 0x13
+	kflag r0
 	bl	secondary_main
